src/sentry/plugins/providers/integration_repository.py

```diff
@@ -33,37 +34,55 @@ def dispatch(self, request, organization, **kwargs):
         except Exception as e:
             return self.handle_api_error(e)
 
-        try:
-            with transaction.atomic():
-                repo = Repository.objects.create(
-                    organization_id=organization.id,
-                    name=result["name"],
-                    external_id=result.get("external_id"),
-                    url=result.get("url"),
-                    config=result.get("config") or {},
-                    provider=self.id,
-                    integration_id=result.get("integration_id"),
+        repo_update_params = {
+            "external_id": result.get("external_id"),
+            "url": result.get("url"),
+            "config": result.get("config") or {},
+            "provider": self.id,
+            "integration_id": result.get("integration_id"),
+        }
+
+        # first check if there is a repository without an integration that matches
+        repo = Repository.objects.filter(
+            organization_id=organization.id, name=result["name"], integration_id=None
+        ).first()
+        if repo:
+            if self.logger:
```

Comment:
do we care about logging if the provider was `None`?

Reply:
@MeredithAnya It doesn't hurt to log it, especially if we need to debug this later for some reason